src/components/store/store.tsx

```diff
@@ -0,0 +1,223 @@
+    setFilterSelected,
+  ] = useState<FilterOption>(FilterOption.All);
+
+  const recieveTodos = useCallback(
+    (newTodos: Todo[]) => {
+      setTodos(newTodos);
+    },
+    [],
+  );
+
+  const filteredTodos = filterTodos({ todos, filterSelected });
+
+  const resetDeletingTodoIds = () => {
+    setDeletingTodoIds([]);
+  };
+
+  const addDeletingTodoIds = (newTodoIds: number[]) => {
+    setDeletingTodoIds(prevTodoIds => [...prevTodoIds, ...newTodoIds]);
+  };
+
+  const resetHasError = useCallback(
+    () => setHasError(null),
+    [],
+  );
+
+  const setError = useCallback(
+    (errMessage: ErrorOption) => {
+      setHasError(errMessage);
+
+      setTimeout(() => resetHasError(), 3000);
+    },
+    [resetHasError],
+  );
+
+  const updateTodoTitle = (todoId: number, title: string) => {
+    const copyTodos = [...todos];
+
+    const findedTodo = copyTodos.find(todo => todo.id === todoId) || null;
+
+    if (!findedTodo) {
+      return;
+    }
+
+    const indexFindedTodo = copyTodos.indexOf(findedTodo);
+    const updatedTodo = {
+      ...findedTodo,
+      title,
+    };
+
+    copyTodos[indexFindedTodo] = updatedTodo;
+
+    setTodos(copyTodos);
+  };
+
+  const addTodo = (newTodo: Todo) => {
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const removeTodo = (todoId: number) => {
+    setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+  };
+
+  const toggleAllTodoCondition = async () => {
+    const findedTodos = todos.filter(outerTodo => {
+      const uncompletedTodos = todos.some(innerTodo => !innerTodo.completed);
+
+      if (uncompletedTodos) {
+        return !outerTodo.completed;
+      }
+
+      return outerTodo.completed;
+    });
+
+    if (findedTodos.length === 0) {
+      return;
+    }
+
+    try {
+      const updatePromises = findedTodos.map(async findedTodo => {
+        const updatedTodo = await apiClient.updateTodo(findedTodo.id, {
+          ...findedTodo,
+          completed: !findedTodo.completed,
+        });
+
+        return updatedTodo;
+      });
+
+      const updatedTodos = await Promise.all(updatePromises);
+
+      setTodos(currentTodos => {
+        return currentTodos.map(todo => (
+          updatedTodos.find(updatedTodo => updatedTodo.id === todo.id) || todo
+        ));
+      });
+    } catch (error) {
+      setHasError(ErrorOption.UpdateTodoError);
+    }
+  };
+
+  const toggleTodoCondition = async (
+    todoId: number,
+    setIsLoading: (loading: boolean) => void,
+  ) => {
+    const findedTodo = todos.find(todo => todo.id === todoId) || null;
+
+    if (!findedTodo) {
+      return;
+    }
+
+    try {
+      const updatedTodo = await apiClient.updateTodo(todoId, {
+        ...findedTodo,
+        completed: !findedTodo.completed,
+      });
+
+      const copyTodos = [...todos];
+      const indexFindedTodo = copyTodos.indexOf(findedTodo);
+
+      copyTodos[indexFindedTodo] = updatedTodo;
+
+      setTodos(copyTodos);
```

Comment:
```suggestion
      setTodos(currentTodos => currentTodos.map(todo => todo.id === todoId ? updatedTodo : todo);
```
Use previous state and `map` instead. Example: